src/codegen/utilities/enum_helpers.py

```diff
@@ -0,0 +1,283 @@
+        match = name_pattern.fullmatch(enum_name)
+        if match:
+            basename = match.group(1)
+
+            if basename in ENUMS_BLACKLIST and enum_name in enums.keys():
+                enums.pop(enum_name) 
+
+            if basename in enums.keys():
+                if basename not in enum_merge_set.keys():
+                    enum_merge_set[basename] = []
+                    enum_merge_set[basename].append(basename)  
+                if  enum_name not in enum_merge_set[basename]:
+                    enum_merge_set[basename].append(enum_name)    
+
+
+    for basename, enums_to_merge in enum_merge_set.items():
+        _logger.debug(f"merging enums: {basename} <-- {enums_to_merge}")
+        enums[basename] = {
+            'values': _merge_enum_values([enums[enum]['values'] for enum in enums_to_merge])
+        }
+        # delete the variants, now
+        for enum in enums_to_merge:
+            if not enum == basename:
+                del enums[enum]
+
+    # sort it by key (enum name)
+    return dict(sorted(enums.items()))
+
+
+def _sanitize_values(enums):
+    for _, enum in enums.items():
+        for value in enum['values']:
+            value_name = value['name']
+    
+            # For Impedance1
+            if value_name in IMPEDANCE_ENUM_VALUE_NAME_SUB.keys():
+                value_name = IMPEDANCE_ENUM_VALUE_NAME_SUB[value_name]
+                
+            for old, new in NAME_SUBSTITUTIONS.items():
+                value_name = value_name.replace(old, new)
+            value['name'] = value_name
+    return enums
+
+
+def get_enums(metadata):
+    enums = metadata['enums']
+
+    # First remove enums we don't use.
+    enums = {name: val for (name, val) in enums.items() if name not in ENUMS_BLACKLIST}
+    # Then merge variants.
+    enums = _merge_enum_variants(enums)
+    return _sanitize_values(enums)
+
+
+def get_enum_value_docstring(enum_value,codegen_metadata):
+    if 'documentation' in enum_value and 'description' in enum_value['documentation']:
+        raw_docstring = get_attr_enum_docstring(enum_value['documentation']['description'],codegen_metadata)
+        raw_docstring = cleanup_docstring(raw_docstring)
+
+        if raw_docstring != '':
+            return f"  #: {raw_docstring}"
+    return ""
+
+def cleanup_docstring(docstring):
+    # Removing leading/trailing whitespace.
+    stripped = docstring.strip()
+
+    # Some strings have extraneous spaces between words; clean those up.
+    words = stripped.split()
+    return " ".join(words)
+
+def get_attr_enum_docstring(docstring, codegen_metadata):
+
+    # Replace occurrences of &attr(PropertyId); with correct property
+    # names.
+    # Replace occurrences of &DAQmx(FunctionName) with correct name.
+    docstring = re.sub(
+        "&DAQmx([^;]+);",
+        lambda m: "DAQmx " + m.group(1),
+        docstring,
+    )
+
+    html_parser = HTMLParser()
+    docstring = html_parser.unescape(docstring)
```

Comment:
`HTMLParser.unescape` was removed in Python 3.9.

```suggestion
    docstring = html.unescape(docstring)
```

Reply:
Updated.